src/Altinn.App.Core/Helpers/Serialization/ModelDeserializer.cs

```diff
@@ -61,50 +58,74 @@ public ModelDeserializer(ILogger logger, Type modelType)
                 return await DeserializeXmlAsync(stream);
             }
 
-            Error = $"Unknown content type {contentType}. Cannot read the data.";
-            return null;
+            return ModelDeserializerResult.FromError($"Unknown content type {contentType}. Cannot read the data.");
         }
 
-        private async Task<object?> DeserializeJsonAsync(Stream stream)
+        private async Task<ModelDeserializerResult> DeserializeMultipartAsync(Stream stream, string contentType)
         {
-            Error = null;
+            MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse(contentType);
+            string boundary = mediaType.Boundary.Value!.Trim('"');
+            var reader = new MultipartReader(boundary, stream);
+            MultipartSection? firstSection = await reader.ReadNextSectionAsync();
+            if (firstSection?.ContentDisposition?.Contains("name=\"dataModel\"") != true)
+            {
+                return ModelDeserializerResult.FromError("First entry in mulipart serialization must have name=\"dataModel\"");
+            }
+            var modelResult = await DeserializeJsonAsync(firstSection.Body);
+            if (modelResult.HasError)
+            {
+                return modelResult;
+            }
 
+            MultipartSection? secondSection = await reader.ReadNextSectionAsync();
+            Dictionary<string, string?>? reportedChanges = null;
+            if (secondSection is not null)
+            {
+                if (secondSection?.ContentDisposition?.Contains("name=\"previousValues\"") != true)
+                {
+                    return ModelDeserializerResult.FromError("First entry in mulipart serialization must have name=\"dataModel\"");
```

Comment:
Missleading error returned
```suggestion
                    return ModelDeserializerResult.FromError("Second entry in mulipart serialization must have name=\"previousValues\"");
```